Source/lighting.cpp

```diff
@@ -757,10 +757,10 @@ void DoVision(int nXPos, int nYPos, int nRadius, BOOL doautomap, BOOL visible)
 					if (!nBlockTable[dPiece[x1adj + nCrawlX][y1adj + nCrawlY]]
 					    || !nBlockTable[dPiece[x2adj + nCrawlX][y2adj + nCrawlY]]) {
 						if (doautomap) {
-							if (dFlags[nCrawlX][nCrawlY] >= 0) {
+							if (!(dFlags[nCrawlX][nCrawlY] & BFLAG_EXPLORED)) {

```

Comment:
why this change?

Reply:
because that is how it works. It checks for that flag, which in this case can be simplified to a sign check, but if BFLAG_EXPLORED is ever changed, the code is messed up.

Reply:
Because the decompiler spat out a signed type which 0x80 translates to `< 0`. Duh.

Reply:
While correct the this not bin exact so probably not how it was written.
![image](https://user-images.githubusercontent.com/204594/95670229-d57b0880-0b88-11eb-8aa0-51bb22aabbea.png)